### PR TITLE
feat: deploy without cni

### PR DESCRIPTION
**Pull request by sergelogvinov**
Run TalosCCM Deployment Without CNI Readines.
When the useDaemonSet parameter is set to true, CCM will use the node IP to access Kubernetes and Talos API.

# Pull Request

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [ ] you ran conformance (`make conformance`)
- [ ] you linted your code (`make lint`)
- [ ] you linted your code (`make unit`)

> See `make help` for a description of the available targets.

### CODE REVIEW

File: charts/talos-cloud-controller-manager/templates/deployment.yaml [hunk 1]
```diff
@@ -65,6 +65,12 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: status.podIP
+            - name: KUBERNETES_SERVICE_HOST
```
Comment (frezbo) [inline charts/talos-cloud-controller-manager/templates/deployment.yaml:68]: isn't it easier to use hostnetwork?

Comment (sergelogvinov) [reply charts/talos-cloud-controller-manager/templates/deployment.yaml:68]: KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT are using for go-kubernetes client connection.

Node can have IPv4 or IPv6 and `status.podIP` will have preferred IP to reach the node. 
I am not sure about 127.0.0.1 or ::1 addresses.

What do you think?

Comment (frezbo) [reply charts/talos-cloud-controller-manager/templates/deployment.yaml:68]: If it's better I'm all for it :thumbsup

**Review by frezbo — APPROVED** [review]

**Pull request merged**